tests/queries/0_stateless/02231_buffer_aggregate_states_leak.sql

```diff
@@ -29,7 +29,7 @@ from in_02231
 group by key;
 
 set optimize_trivial_insert_select = 1;
-insert into in_02231 select * from numbers(10e6) settings max_memory_usage='400Mi', max_threads=1;
+insert into in_02231 select * from numbers(5e6) settings max_memory_usage='40Mi', max_threads=1, min_insert_block_size_rows = 10000, min_insert_block_size_bytes = '10Mi';
```

Comment:
a. This new test does not shows the original problem anymore, because you limited the block size, you can verify the test by reverting changes related to StorageBuffer from e2960e1a524afb5c0341a121748d10cae938d797

b. Tests will small `max_memory_usage` tends to be unstable due to memory tracking is not that precise

Reply:
How to make a better test then?

Reply:
Ok, now more conservative changes.